src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -954,40 +954,44 @@ int TPCHandler::ProcessPullReq(const std::string &resource, XrdHttpExtReq &req)
             return req.SendSimpleResp(rec.status, NULL, NULL, msg, 0);
     }
     // ddavila 2023-01-05:
-    // Force the TPC to use the IP, that was used to contact the server,
-    // as the Destination address.
-
-    XrdNetAddr *nP;
-    int numIP = 0;
-    char buff[1024];
-    char * ip;
-
-    // Get the hostname used to contact the server from the http header
-    auto host_header = req.headers.find("Host");
-    std::string host_used;
-    if (host_header != req.headers.end()) {
-        host_used = host_header->second;
-    }
-
-    // Get all the IP addresses associated with the above hostname
-    XrdNetUtils::GetAddrs(host_used.c_str(), &nP, numIP, XrdNetUtils::prefAuto, 0);
+    // The following change was required by the Rucio/SENSE project where
+    // multiple IP addresses, each from a different subnet, are assigned to a
+    // single server and routed differently by SENSE.
+    // The above requires the server to utilize the same IP, that was used to
+    // start the TPC, for the resolution of the given TPC instead of
+    // using any of the IPs available.
+    char *sense = getenv( "XRD_SENSE" );
```

Comment:
I apologize for note getting to you earlier about this. An alternative is to add a config directive "tpc.sense" no need for an argument as it's implied to be true. This can trivially be done in TPCHandler::Configure @ XrdTpcConfigure.cc.  You'll see that you will be able to set a bool in XrdTpcTpc.hh which can be tested here. The place to do it will be obvious (e.g. before/after the test for tpc.timeout). It's a trivial addition and avoids proliferating global envars. OK?